README.md

```diff
@@ -13,12 +13,65 @@ helm plugin install https://github.com/true-north-engineering/helm-file-utils
 
 ## Usage
 
-This is only applicable to the `-f` or `--values` option of a Helm
+Helm File Utils allows user to do multiple transformations over given file. 
+This plugin is only applicable to the `-f` or `--values` option of a Helm
 command (e.g. `install`, `upgrade` or `template`).  The basic usage
 is to reference a directory (either absolutely, or relative to the
 PWD) from which to collect all non-hidden files with the extension
-`.yaml` or `.yml`, not including sub-directories:
+`.yaml` or `.yml`, not including sub-directories. Keyword used to 
+associate plugin with given file directory is **futl**.
+
+Basic usage of plugin is as it follows:
+````bash
+helm install -f futl://path/to/yaml/file /path/to/chart/file
+````
+
+### File transformations
+
+In given `.yaml` or `.yml` file, multiple file transformations are possible.
+Transformations are classified in two categories - Transformers(**T**) and Readers(**R**). 
+Every command needs to consist of **at most** one Reader and **at least** one Transformer separated with **+** sign.
+Order of transformation evaluation is from right to left, which forces Reader to always execute first.
+
+**Reader**\
+Used to read the file from given destination. If none is provided, **file** is considered as default.\
+Available Readers: **file, dir, ~~http(s)~~, ~~git~~**
+
+**Transformer**\
+Transformers are used to do various transformations over the file.\
+Available Transformers: **base64enc, ~~base64dec, xslt, custom~~**
+
+Template for chaining file transformations is:
+````bash
+!futl T+T+T+R://path/to/file
+````
+
+
+##Example usage
+
+````bash
+helm install -f futl://home/usr/files /home/usr/charts
```

Comment:
Can you please provide here a valid command.  [Helm install](https://helm.sh/docs/helm/helm_install/) probably needs a chart name

Reply:
For helm CLI please always use generic examples, e.g. -
helm install [NAME] [CHART] [flags] -f futl://path/to/values.yaml